### PR TITLE
Publish allennlp-server to PyPI and misc CI changes

### PR DESCRIPTION
**Pull request by agururajvais**
Changes for Issue: https://github.com/allenai/allennlp/issues/4974

Acknowledgment of support by @pab-vmware , @prajaktakini-vmware, @annajung, @divijbajaj

### CODE REVIEW

File: README.md
Comment (pab-vmware) [inline README.md:23]: Misleading unless you can point to the allenlp-server instance itself.

File: setup.py
Comment (pab-vmware) [inline setup.py:44]: Not sure if this is applicable. 
Probably Dirk can confirm.

File: README.md
Comment (pab-vmware) [inline README.md:43]: It requires both allennlp core and the models package

File: .github/workflows/main.yml
Comment (pab-vmware) [inline .github/workflows/main.yml:226]: github indicates a missing EOF newline

**Comment by gururajvaishnav** [comment]
> Changes for Issue: [allenai/allennlp#4974](https://github.com/allenai/allennlp/issues/4974)

Acknowledgment of support by` @pab-vmware, https://github.com/prajaktakini-vmware https://github.com/annajung https://github.com/divijbajaj `

**Comment by gururajvaishnav** [comment]
@dirkgr I am contributing as VMware and need to get approval before merging, and is currently in process of getting the repo approved. Can you review and if all is good, can we wait for a couple of days before we can merge the changes? :D

**Comment by agururajvais** [comment]
@dirkgr  I am contributing as VMware and need to get approval before merging, as it is currently in process of getting my contribution approved. Can you review and if all is good, can we wait for a couple of days before we can merge the changes? :D

**Comment by gururajvaishnav** [comment]
> Changes for Issue: [allenai/allennlp#4974](https://github.com/allenai/allennlp/issues/4974)

Acknowledgment of support by @pab-vmware, @prajaktakini-vmware @annajung @divijbajaj

File: setup.py
Comment (dirkgr) [reply setup.py:44]: I don't think we can say this. This doesn't really visualize the output (though we keep talking about doing that at some point ...).

**Review by dirkgr — COMMENTED** [review]
This looks good to me, but I'd rather have @epwalsh take a look. He did the publishing pipeline for the rest of AllenNLP.

File: .github/dependabot.yml
Comment (epwalsh) [inline .github/dependabot.yml:10]: Why is this needed?

File: .github/workflows/main.yml
Comment (epwalsh) [inline .github/workflows/main.yml:63]:
```suggestion
      if: github.repository == 'allenai/allennlp-server' && (github.event_name == 'push' || github.event_name == 'pull_request')
```

File: .github/workflows/main.yml
Comment (epwalsh) [inline .github/workflows/main.yml:89]: Let's be consistent with 3.8:

```suggestion
          python-version: 3.8
```

File: .github/workflows/main.yml
Comment (epwalsh) [inline .github/workflows/main.yml:159]: How about just not checking out the code in the first place?

```suggestion
```

File: .github/workflows/main.yml
Comment (epwalsh) [inline .github/workflows/main.yml:211]:
```suggestion
          python-version: 3.8
```

File: .github/workflows/main.yml
Comment (epwalsh) [inline .github/workflows/main.yml:207]: Probably don't need to checkout here either.

```suggestion
```

**Review by epwalsh — CHANGES_REQUESTED** [review]
Hey @agururajvais, this looks good, thanks for taking this on. Just a few comments.

Oh, and can you remove the `setup.cfg` file?

File: setup.py
Comment (agururajvais) [reply setup.py:44]: Cool. Will remove this.

File: .github/dependabot.yml
Comment (agururajvais) [reply .github/dependabot.yml:10]: The whole of dependabot or just nr-databind-core? I used the same config from allennlp to keep things consistent.

Comment (epwalsh) [reply .github/dependabot.yml:10]: Ah, gotcha. We definitely want dependabot, so thanks for adding that. But I don't think we need this `ignore` section for `nr-databind-core`.

File: README.md
Comment (epwalsh) [inline README.md:38]:
```suggestion
Note that the `allennlp-server` package is tied to the [`allennlp` core package](https://pypi.org/projects/allennlp) and [`allennlp-models` package](https://pypi.org/projects/allennlp-models). Therefore when you install the server package you will get the latest compatible version of `allennlp` and `allennlp-models` (if you haven't already installed `allennlp` or `allennlp-models`). For example,
```

**Review by epwalsh — APPROVED** [review]
LGTM! Thanks so much!

I'll trigger a release tomorrow so we get this on PyPI.

**Pull request merged**
